src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs

```diff
@@ -66,7 +66,11 @@ protected virtual BlobContainerClient CreateBlobContainerClient(string connectio
             if (_azureBlobStorageProvider.TryCreateBlobServiceClientFromConnection(connection, out BlobServiceClient blobServiceClient))
             {
                 var blobContainerClient = blobServiceClient.GetBlobContainerClient(_secretsContainerName);
-                blobContainerClient.CreateIfNotExists();
+
+                if (!blobContainerClient.Exists())
```

Comment:
Documentation of CreateIfNotExists says it does an existence check, so adding this additional existence check means we're checking twice, then doing the create operation when necessary?

Also, in https://github.com/Azure/azure-sdk-for-net/issues/109#issuecomment-1288786252 it's mentioned that this Exists check itself is also generates a warning "404 The specified container does not exist.". Is that true?

Reply:
Yeah I don't love it but I'm not sure how else to address this given that that CreateIfNotExists with throw (catching won't help because App Insights will still log the operation).

As for the 404, I'm not sure I'll need to test it out myself

Reply:
I believe that the `CreateIfNotExists()` method does an attempt to create -- and silently handles the 409 conflict internally. This means you almost always get a 409 from this once the table is created. App Insights can see this and reports it as an error.

Switching to the `Exists()` check -- you'll get a 404 once, and then, generally, you'll never see it again as the table will always exist and you'll get a 200 from then on.

This has been an issue elsewhere and we fix them as we get complaints. I added this here a while back b/c we got the same complaints: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs#L515-L520